### PR TITLE
Fix/sg upgrade testing

### PR DESCRIPTION
**Pull request by euniceatcouchbase**
#### Fixes #.

- [X] Ran `flake8`
- [X] Ran `run_repo_tests.sh`

#### Changes proposed in this pull request:

- added missing parameters in sg config file during upgrade
- set use_views value dynamically based on cb server version
- added an utility method to pull cb server version
- added condition check in ansible script in order to conform various cb server version cli command

### CODE REVIEW

File: keywords/SyncGateway.py [hunk 1]
```diff
@@ -561,11 +578,16 @@ def upgrade_sync_gateways(self, cluster_config, sg_conf, sync_gateway_version, u
                 log_info("Keyerror in getting logging{}".format(ex.message))
                 playbook_vars["logging"] = '{} {},'.format(logging_config, "}")
 
-            if get_sg_use_views(cluster_config):
-                playbook_vars["sg_use_views"] = '"use_views": true,'
+            server_version = get_cbs_version(cluster_config)
+            cbs_version, cbs_build = version_and_build(server_version)
+            if cbs_version >= "5.5.0":
+                if get_sg_use_views(cluster_config):
+                    playbook_vars["sg_use_views"] = '"use_views": true,'
```
Comment (sridevi-15) [inline keywords/SyncGateway.py]: It is same logic even for else. you can combine the logic

File: libraries/provision/ansible/playbooks/install-couchbase-server-package.yml [hunk 2]
```diff
@@ -128,39 +127,73 @@
       shell: "{{ couchbase_server_home_path }}/bin/couchbase-cli node-init -c {{ couchbase_server_node }}:{{ couchbase_server_admin_port }} --user={{ couchbase_server_admin }} --password={{ couchbase_server_password }} --node-init-hostname={{ couchbase_server_node }}"
       when: "'3.0' in couchbase_server_package_name or '3.1' in couchbase_server_package_name"
 
-    - name: COUCHBASE SERVER | Initialize primary node | configure IPv4
+    - name: COUCHBASE SERVER | Initialize primary node (4.X and 5.X) | configure IPv4
+      shell: "{{ couchbase_server_home_path }}/bin/couchbase-cli node-init -c {{ couchbase_server_node }}:{{ couchbase_server_admin_port }} --user={{ couchbase_server_admin }} --password={{ couchbase_server_password }} --node-init-hostname={{ couchbase_server_node }}"
+      when: "{{ cb_major_version['stdout'] | search('[4-5]')}} and not ipv6_enabled"
```
Comment (sridevi-15) [inline libraries/provision/ansible/playbooks/install-couchbase-server-package.yml]: can you verify that major version is 15 or 14, it still skips and this condition applies only for 4. or 5. , but not for 15. or 14.

File: keywords/SyncGateway.py [hunk 3]
```diff
@@ -561,11 +578,16 @@ def upgrade_sync_gateways(self, cluster_config, sg_conf, sync_gateway_version, u
...
+                else:
+                    num_replicas = get_sg_replicas(cluster_config)
+                    playbook_vars["num_index_replicas"] = '"num_index_replicas": {},'.format(num_replicas)
             else:
-                num_replicas = get_sg_replicas(cluster_config)
-                playbook_vars["num_index_replicas"] = '"num_index_replicas": {},'.format(num_replicas)
+                playbook_vars["sg_use_views"] = '"use_views": true,'
```
Comment (hrajput89) [inline keywords/SyncGateway.py]: I'm not sure if SG versions <5.5 recognises "use_views" as property. Can you check with Ben or Sri, regarding same? As far as I remember, use_views was added as to 5.5 for backward compatibility.

Comment (euniceatcouchbase) [reply keywords/SyncGateway.py]: comments from Adam: 
SG 2.1 uses GSI by default, which requires CBS 5.5 or later.  The use_views flag was included in SG 2.1 to support running w/ older versions of CBS (by setting use_views=true)
full details here https://docs.couchbase.com/sync-gateway/2.1/compatibility-matrix.html

File: keywords/SyncGateway.py (hunk 1 above)
Comment (euniceatcouchbase) [reply keywords/SyncGateway.py]: checked in a new copy, combined the logic

File: libraries/provision/ansible/playbooks/install-couchbase-server-package.yml (hunk 2 above)
Comment (euniceatcouchbase) [reply libraries/provision/ansible/playbooks/install-couchbase-server-package.yml]: tuned regex pattern to check single digit number if version is 4.x or 5.x and check single digit + not between 1 to 5 if version is 6.x or above

File: keywords/SyncGateway.py (hunk 1 above)
Comment (euniceatcouchbase) [reply keywords/SyncGateway.py]: done

File: libraries/provision/ansible/playbooks/install-couchbase-server-package.yml (hunk 2 above)
Comment (euniceatcouchbase) [reply libraries/provision/ansible/playbooks/install-couchbase-server-package.yml]: done

**Pull request merged**